packages/track-component/index.js

```diff
@@ -156,17 +156,27 @@ class Track extends Component {
       share: true
     }
 
-    return menuButton.render({
-      items: [], // no custom items yet
-      selection: Object.entries(selection).filter(([k, v]) => Boolean(v)).map(([k, v]) => k), // selection to array of keys
-      data: Object.assign({}, this.local.track, {
-        count: this.local.count,
-        favorite: this.local.favorite || this.local.fav,
-        url: new URL(`/track/${this.local.track.id}`, process.env.APP_HOST || 'https://stream.resonate.coop')
-      }),
-      size: this.local.type === 'album' ? 'sm' : 'md', // button size
-      orientation: 'bottomright'
-    })
+    const iOS = (navigator !== undefined && 'userAgent' in navigator) &&
```

Comment:
Between these two PRs, should this be a re-usable function?

Reply:
Great idea @simonv3.

I get this error:
```
◇ Error: Can't walk dependency graph: Cannot find module 'beta/src/helpers/isMobile' from '/Users/peterklingelhofer/Dev/stream/beta/build/views/settings/index.js'
    required by /Users/peterklingelhofer/Dev/stream/beta/build/views/settings/index.js
    at /Users/peterklingelhofer/Dev/stream/node_modules/resolve/lib/async.js:146:35
    at processDirs (/Users/peterklingelhofer/Dev/stream/node_modules/resolve/lib/async.js:299:39)
    at isdir (/Users/peterklingelhofer/Dev/stream/node_modules/resolve/lib/async.js:306:32)
    at /Users/peterklingelhofer/Dev/stream/node_modules/resolve/lib/async.js:34:69
    at FSReqCallback.oncomplete (fs.js:191:21)
```

Using this diff:
```diff
diff --git a/beta/src/views/settings/index.js b/beta/src/views/settings/index.js
index 568247f..71415c9 100644
--- a/beta/src/views/settings/index.js
+++ b/beta/src/views/settings/index.js
@@ -1,6 +1,7 @@
 const html = require('choo/html')
 const button = require('@resonate/button')
 const viewLayout = require('../../layouts/default')
+const isIOS = require('../../helpers/isMobile')
 
 module.exports = () => viewLayout(renderSettings)
 
@@ -20,12 +21,7 @@ function renderSettings (state, emit) {
     size: 'none'
   })
 
-  const cookieConsentGranted = state.cookieConsentStatus === 'allow'
-  const iOS = (navigator !== undefined && 'userAgent' in navigator) &&
-    (navigator.userAgent.match(/iOS/) ||
-    (navigator.userAgent.match(/Mac/) && document !== undefined && 'ontouchend' in document))
-
-  const notificationsSection = html`
+  const notificationSettings = html`
     <div class="bg-light-gray bg-light-gray--light bg-transparent--dark ba b--gray b--gray--light 
b--near-black--dark mb3 pa3">
       <fieldset class="pa0 ma0 bn">
         <legend class="f3 lh-title ma0 pa0">Notifications</legend>
@@ -37,7 +33,8 @@ function renderSettings (state, emit) {
     </div>
   `
 
-  const cookieConsentSection = html`
+  const cookieConsentGranted = state.cookieConsentStatus === 'allow'
+  const cookieConsentSettings = html`
     <div class="bg-light-gray bg-light-gray--light bg-transparent--dark ba b--gray b--gray--light 
b--near-black--dark mb3 pa3">
       <fieldset class="pa0 ma0 bn">
         <legend class="f3 lh-title ma0">Cookie consent</legend>
@@ -59,8 +56,8 @@ function renderSettings (state, emit) {
 
   return html`
     <section id="app-settings" class="flex flex-column ph4 mt3">
-      ${!iOS ? notificationsSection : ''}
-      ${cookieConsentSection}
+      ${!isIOS ? notificationSettings : ''}
+      ${cookieConsentSettings}
     </section>
   `
 }
```

Originally tried importing it as a TypeScript file but no dice with that either, same error, and with both import and require, destructured and not. Can you spot anything immediately wrong with my import/require statement?